### PR TITLE
Data migration unit test

### PR DESCRIPTION
**Pull request by jonescc**
Rename transformCatalogue to follow java naming conventions for classes.
Add example unit test for testing data migration.

@APaulic - test metadata validates.  Haven't added missing elements to expected result/got it to validate.

Had to turn off autoformat of xml files (code style settings) to stop intellij reformatting my expected result).

### CODE REVIEW

**Comment by APaulic** [comment]
Added both valid MCP and 19115-3 records to complete test coverage

**Comment by APaulic** [comment]
Requires an update to test:
1. an empty MD_Scope object
1. metadataConstraints/MD_Commons
1. Confirm all mappings tested

**Comment by APaulic** [comment]
Unit test does not cover replacement of codeListLocation. That issue should be addressed as part of https://github.com/aodn/backlog/issues/1568

**Comment by jonescc** [comment]
Actually, this is one I'd like to review.  Its a key mapping pull request.

**Comment by jonescc** [comment]
Sachit can DOD

**Comment by jonescc** [comment]
Or review as well.  A fresh perspective would be good.  But I'd like to confirm mapping tests are as required.

**Comment by jonescc** [comment]
As discussed - would be good to add some of the missed tags as an example unit test for when we actually fix issues to finish this PR of.

[All tags used in full export](https://github.com/aodn/utilities/files/4390403/all-tags.txt)
[Currently tested tags.txt](https://github.com/aodn/utilities/files/4390404/test-tags.txt)

[diff.txt](https://github.com/aodn/utilities/files/4390412/diff.txt)

**Comment by jonescc** [comment]
Note my dump is before Sachit's validation fixes.  I note there are a couple of elements include that will be removed . e.g.:

```
mcp:MD_Metadata/gmd:describes
/mcp:MD_Metadata/gmd:series
```

And probably some others.

**Comment by jonescc** [comment]
@sachitrajbhandari can you approve - I opened the pull request so can't approve it.

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/expected/f48531e2-f182-56ca-e043-08114f8c7f2e.xml [hunk 1]
```diff
@@ -0,0 +1,4419 @@
...
...o/19115/-3/mrc/2.0 http://standards.iso.org/iso/19115/-3/mrc/2.0/mrc.xsd http://standards.iso.org/iso/19115/-3/mrd/1.0 http://standards.iso.org/iso/19115/-3/mrd/1.0/mrd.xsd http://standards.iso.org/iso/19115/-3/mri/1.0 http://standards.iso.org/iso/19115/-3/mri/1.0/mri.xsd http://standards.iso.org/iso/19115/-3/mrl/2.0 http://standards.iso.org/iso/19115/-3/mrl/2.0/mrl.xsd http://standards.iso.org/iso/19115/-3/mrs/1.0 http://standards.iso.org/iso/19115/-3/mrs/1.0/mrs.xsd http://standards.iso.org/iso/19115/-3/msr/2.0 http://standards.iso.org/iso/19115/-3/msr/2.0/msr.xsd http://standards.iso.org/iso/19157/-2/mdq/1.0 http://standards.iso.org/iso/19157/-2/mdq/1.0/mdq.xsd http://standards.iso.org/iso/19115/-3/mac/2.0 http://standards.iso.org/iso/19115/-3/mac/2.0/mac.xsd http://standards.iso.org/iso/19115/-3/gco/1.0 http://standards.iso.org/iso/19115/-3/gco/1.0/gco.xsd http://standards.iso.org/iso/19115/-3/gmw/1.0 http://standards.iso.org/iso/19115/-3/gmw/1.0/gmw.xsd http://www.opengis.net/gml/3.2 http://schemas.opengis.net/gml/3.2.1/gml.xsd http://www.w3.org/1999/xlink http://www.w3.org/1999/xlink.xsd http://schemas.aodn.org.au/mcp-3.0 http://schemas.aodn.org.au/mcp-3.0/mcpExtensions.xsd">
```
Comment (sachitrajbhandari) [inline geonetwork/transform-mcp-iso19115-3/src/test/resources/expected/f48531e2-f182-56ca-e043-08114f8c7f2e.xml]: MCP-3.0 Profile removed from transform with PR https://github.com/aodn/utilities/pull/205. So these need to be removed from expected files.
```
xmlns:mcp="http://schemas.aodn.org.au/mcp-3.0
...
http://schemas.aodn.org.au/mcp-3.0 http://schemas.aodn.org.au/mcp-3.0/mcpExtensions.xsd
```

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/expected/f48531e2-f182-56ca-e043-08114f8c7f2e.xml [hunk 2]
```diff
@@ -0,0 +1,4419 @@
...
+         </mri:abstract>
+         <mri:credit>
```
[the remark this answers is not in the document]
Comment (APaulic) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/expected/f48531e2-f182-56ca-e043-08114f8c7f2e.xml]: Is it currently being mapped incorrectly or not at all? The mapping is in backlog item https://github.com/aodn/backlog/issues/1647 TBA

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/expected/f48531e2-f182-56ca-e043-08114f8c7f2e.xml (hunk 1 above)
Comment (APaulic) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/expected/f48531e2-f182-56ca-e043-08114f8c7f2e.xml]: Fixed

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/expected/f48531e2-f182-56ca-e043-08114f8c7f2e.xml [hunk 3]
```diff
@@ -0,0 +1,4419 @@
...
+      </cit:CI_Date>
+   </mdb:dateInfo>
+   <mdb:metadataStandard>
+      <cit:CI_Citation>
+         <cit:title>
+            <gco:CharacterString>ISO 19115-3:2018</gco:CharacterString>
+         </cit:title>
+      </cit:CI_Citation>
+   </mdb:metadataStandard>
+   <mdb:metadataProfile>
```
[the remark this answers is not in the document]
Comment (APaulic) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/expected/f48531e2-f182-56ca-e043-08114f8c7f2e.xml]: Fixed

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/expected/2d2b2f92-12fa-4330-a480-94f0892c2b72.xml [hunk 4]
```diff
@@ -0,0 +1,2140 @@
...
+               </mcc:level>
+            </mcc:MD_Scope>
+         </mac:scope>
+         <mac:instrument>
+            <mac:MI_Instrument>
+               <mac:identifier>
+                  <mcc:MD_Identifier>
+                     <mcc:code>
+                        <gcx:Anchor xlink:href="http://vocab.nerc.ac.uk/collection/L05/current/22">plankton nets</gcx:Anchor>
+                     </mcc:code>
```
[the remark this answers is not in the document]
Comment (APaulic) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/expected/2d2b2f92-12fa-4330-a480-94f0892c2b72.xml]: Fixed. Changed both exported and expected files.

See https://github.com/aodn/utilities/pull/201/commits/5a27ffcb31aff1f271b0bffa33b3ac1cf56aa387

**Comment by jonescc** [comment]
Hey guys, we aren't meant to be fixing mapping issues in this PR.  If there's an outstanding mapping issue, we'll fixup the unit tests when we fix that issue.  If there's an issue which hasn't been raised yet.  Lets raise it.

So we can merge this PR as long as any issues are captured elsewhere.

**Comment by jonescc** [comment]
@sachitrajbhandari - can this be merged now?

**Comment by sachitrajbhandari** [comment]
@jonescc Reviewed and ready to merge

**Pull request merged**